feat: replace automatic mixed precision with explicit precision

cathalobrien opened the pull request:
## Description

This PR replaces the use of automatic precision with explicit precision. 

## Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

<!-- Link the Issue number this change addresses, ideally in one of the "magic format" such as Closes #XYZ -->

<!-- Alternatively, explain the motivation behind the changes and the context in which they are being made. -->

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I ran the [complete Pytest test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html) suite locally, and they pass
-   [x] I have tested the changes on a single GPU
-   [x] I have tested the changes on multiple GPUs / multi-node setups
-   [ ] I have run the Benchmark Profiler against the old version of the code
-   [ ] If the new feature introduces modifications at the config level, I have made sure to update Pydantic Schemas and default configs accordingly


<!-- In case this affects the model sharding or other specific components please describe these here. -->

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ ] I have not introduced new dependencies in the inference portion of the pipeline

<!-- List any new dependencies that are required for this change and the justification to add them. -->

### Documentation

-   [ ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ ] I have added comments to my code, particularly in hard-to-understand areas

<!-- Describe any major updates to the documentation -->

## Additional Notes

<!-- Include any additional information, caveats, or considerations that the reviewer should be aware of. -->


<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚: https://anemoi-training--259.org.readthedocs.build/en/259/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚: https://anemoi-graphs--259.org.readthedocs.build/en/259/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚: https://anemoi-models--259.org.readthedocs.build/en/259/

<!-- readthedocs-preview anemoi-models end -->

cathalobrien committed:
replace automatic mixed precision with explict precision

cathalobrien committed:
replace automatic mixed precision with explict precision

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mchantry commented on models/src/anemoi/models/layers/mapper.py:
```diff
@@ -139,7 +139,12 @@ def _register_edges(
         assert edge_attributes is not None, "Edge attributes must be provided"
 
-        edge_attr_tensor = torch.cat([sub_graph[attr] for attr in edge_attributes], axis=1)
+        # explicit cast is needed here to stay in fp16 precision. default_dtype is set in models/encoder_processor_decoder.py
```
Does this have an impact on skill?

mchantry commented on models/src/anemoi/models/models/encoder_processor_decoder.py:
```diff
@@ -76,7 +76,17 @@ def __init__(
         self.num_channels = model_config.model.num_channels
 
-        self.node_attributes = NamedNodesAttributes(model_config.model.trainable_parameters.hidden, self._graph_data)
+        precision_mapping = {
```
Can we describe the torch object in the hydra config directly? save creating this dict?

mchantry commented:
Cool contribution.
Do we have assessment about whether this has no impact on model skill?

dietervdb-meteo committed:
feat: edge post-processor (#199)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Mario Santa Cruz <48736305+JPXKQX@users.noreply.github.com>

jakob-schloer committed:
docs: Update docs for kcrps. (#258)

Co-authored-by: Simon Lang <simon.lang@ecmwf.int>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: anaprietonem <ana.prietonemesio@ecmwf.int>
Co-authored-by: Matthew Chantry <matthew.chantry@ecmwf.int>
Co-authored-by: Mario Santa Cruz <48736305+JPXKQX@users.noreply.github.com>
Co-authored-by: Harrison Cook <Harrison.cook@ecmwf.int>

anaprietonem committed:
fix: dry run forking (#260)

Co-authored-by: Julien Lefaucheur <jjlk@users.noreply.github.com>

cathalobrien commented:
I'll start some training runs and check

anaprietonem committed:
fix: pydantic schemas move (#228)

Co-authored-by: theissenhelen <helen.theissen@ecmwf.int>

DeployDuck committed:
chore(main): Release graphs 0.5.1 (#197)

DeployDuck committed:
chore(main): Release models 0.5.0 (#139)

anaprietonem committed:
fix: update to null and bump versions (#263)

DeployDuck committed:
chore(main): Release training 0.4.0 (#126)

jakob-schloer committed:
docs: Fix minor mistakes in CRPS user guide. (#264)

ssmmnn11 commented on models/src/anemoi/models/models/encoder_processor_decoder.py:
```diff
@@ -76,7 +76,17 @@ def __init__(
+
+        self.node_attributes = NamedNodesAttributes(
+            model_config.model.trainable_parameters.hidden, self._graph_data
+        ).type(self.dtype)
```
I think here we just handle tensors. If we register them as buffers, doesn't the lightning module take care of the conversion?

ssmmnn11 reviewed (changes requested):
this will be very nice. A couple of things we need to understand better.

JPXKQX committed:
feat(graphs): support edge indices sorting (#187)

JPXKQX committed:
feat(graphs): add scale_resolutions arg (#188)

MeraX committed:
ci: Clean remnants of tests/ directory (#275)

JPXKQX committed:
fix(graphs): rename zarr occurrences in anemoi-graphs (#273)

DeployDuck committed:
chore(main): Release graphs 0.5.2 (#269)

JPXKQX committed:
fix(training): bump anemoi-graphs version to 0.5.2 (#276)

MeraX committed:
test: aicon metadata (#253)

jakob-schloer committed:
fix: Fix inference with kcrps  (#277)

It requires a new config keyword, `model.interface`.

mchantry committed:
fix: Revert "fix: Fix inference with kcrps " (#280)

jakob-schloer committed:
fix: Adapt predict_step in model interface to pass on arguments for model classes (#281)

VeraChristina committed:
ci: fix install mode in pr action (#283)

VeraChristina committed:
ci: fix install mode in pr action (#285)

MeraX committed:
chore: Define minimum versions for all dependencies. (#254)

## Description

Define minimum versions for all dependencies and update some minimal package versions are necessary to pass all tests with lowest package versions. Keeping the dependencies up to date helps when gradually updating Anemoi packages.

## Intent and Issue Number

Keep dependencies up to date.
Closes #266

DeployDuck committed:
chore(main): Release models 0.5.1 (#278)

anaprietonem committed:
fix: clean params complement (#291)

Co-authored-by: Harrison Cook <Harrison.cook@ecmwf.int>

VeraChristina committed:
ci: refactor test pr workflow (#292)

sahahner committed:
fix!: Rework Loss Scalings to provide better modularity (#52)

Co-authored-by: Mario Santa Cruz <mariosanta_cruz@hotmail.com>
Co-authored-by: Mariana Clare <mariana.clare@ecmwf.int>
Co-authored-by: Ewan Pinnington <ewan.pinnington@ecmwf.int>
Co-authored-by: Harrison Cook <harrison.cook@ecmwf.int>
Co-authored-by: jakob-schloer <jakob.schloer@gmail.com>
A big thank you to all the involved reviewers. This would not have been possible without you.

jakob-schloer committed:
fix: Adapt ensemble configs to work with loss refactor.  (#300)

pre-commit-ci[bot] committed:
chore(deps): pre-commit.ci autoupdate (#301)

sahahner committed:
fix(training): no logging variable metadata to mlflow (#304)

mc4117 committed:
fix: Change output format in validation step to fix issue in plotting (#305)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

japols committed:
feat: GraphtransformerProcessor chunking (#66)

Co-authored-by: Harrison Cook <Harrison.cook@ecmwf.int>

VeraChristina committed:
ci: add notify-team step for integration test workflow (#295)

VeraChristina committed:
test: add test for restart of training cycle (#306)

HCookie committed:
fix: Move scaler to device in scale (#317)

theissenhelen committed:
ci: hpc gpu integration test (#236)

This adds a workflow for running integration tests on gpu.

DeployDuck committed:
chore(main): Release models 0.6.0 (#298)

theissenhelen committed:
feat: transformer mapper (#179)

implements transformer mappers as an addition to the GNN and GNNTransformer mapper

Co-authored-by: Simon Lang <simon.lang@ecmwf.int>
Co-authored-by: Rilwan Adewoyin <18564167+Rilwan-Adewoyin@users.noreply.github.com>

JPXKQX committed:
fix(graphs,directions): fix error with direction computation (#287)

DeployDuck committed:
chore(main): Release graphs 0.6.0 (#299)

VeraChristina committed:
ci: run notify-team step only on cron jobs in integration tests (#308)

cathalobrien commented on models/src/anemoi/models/layers/mapper.py:
```diff
@@ -139,7 +139,12 @@ def _register_edges(
+        # explicit cast is needed here to stay in fp16 precision. default_dtype is set in models/encoder_processor_decoder.py
+        edge_attr_tensor = (
+            torch.cat([sub_graph[attr] for attr in edge_attributes], axis=1)
+            .type(torch.float16)
```
my mistake, removed

mc4117 committed:
test: add integration test for the crps ensemble (#325)

DeployDuck committed:
chore: synced file(s) with ecmwf/reusable-workflows (#328)

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Gert Mertes <gert.mertes@ecmwf.int>

HCookie committed:
fix(training): Explicit Batch invariance (#318)

## Description

Add explicit batch normalisation to the loss function

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
-   [ ] Documentation update


## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] I ran the [complete Pytest
test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html)
suite locally, and they pass
-   [ ] I have tested the changes on a single GPU
-   [ ] I have tested the changes on multiple GPUs / multi-node setups
- [ ] I have run the Benchmark Profiler against the old version of the
code
- [ ] If the new feature introduces modifications at the config level, I
have made sure to update Pydantic Schemas and default configs
accordingly


<!-- In case this affects the model sharding or other specific
components please describe these here. -->

### Dependencies

- [x] I have ensured that the code is still pip-installable after the
changes and runs
- [x] I have tested that new dependencies themselves are
pip-installable.
- [x] I have not introduced new dependencies in the inference portion of
the pipeline

<!-- List any new dependencies that are required for this change and the
justification to add them. -->

### Documentation

-   [x] My code follows the style guidelines of this project
- [x] I have updated the documentation and docstrings to reflect the
changes
- [x] I have added comments to my code, particularly in
hard-to-understand areas

<!-- Describe any major updates to the documentation -->

---------

Co-authored-by: Mario Santa Cruz <mariosanta_cruz@hotmail.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Mariana Clare <31656450+mc4117@users.noreply.github.com>

chebertpinard committed:
feat!: generalise activation function (#163)

## Description

We want to make the activation function more flexible to allow adding
those that are not in pytorch. In this situation the GLU derivated
(SwiGLU, ReGLU, GEGLU) would be interesting to test.

There is a pytorch implementation of the GLU function, but it splits the
input in 2 parts and make the output half the size of the input. We then
have issue with input size of the next layer. The suggested solution for
that is to make the linear transformation on the complete initial
matrix, but other options are possible and could replace that one.

```py
W = nn.Linear(dim, dim)
V = nn.Linear(dim, dim)

GLU = Sigmoid(W(input)) * V(input)
```

Equation: GLU(x, W, V, b, c) = σ(xW + b) ⊗ (xV + c)

---------

Co-authored-by: Jesper Dramsch <jesper@dramsch.net>
Co-authored-by: Jesper Dramsch <jesper.dramsch@ecmwf.int>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Ana Prieto Nemesio <91897203+anaprietonem@users.noreply.github.com>

anaprietonem committed:
fix: remove activation entry from MLP noise block (#340)

## Description
<!-- What issue or task does this change relate to? --> 

## What problem does this change solve?
<!-- Describe if it's a bugfix, new feature, doc update, or breaking
change -->

## What issue or task does this change relate to?
error - https://github.com/ecmwf/anemoi-core/actions/runs/15278237765
fix - https://github.com/ecmwf/anemoi-core/actions/runs/15299027451

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

---------

Co-authored-by: Jesper Dramsch <jesper.dramsch@ecmwf.int>

JPXKQX committed:
fix(graphs): fix device mismatch error (#339)

## Description

This PR resolves an issue with mismatches between different devices that
was raised during integration tests using the GPU.

## What problem does this change solve?

Error trace raised by the integration tests.
```
    def get_raw_values(self, nodes: NodeStorage, **kwargs) -> torch.Tensor:
        assert self.mask_node_attr_name in nodes, f"Node attribute '{self.mask_node_attr_name}' not found in nodes."
        attr_values = super().get_raw_values(nodes, **kwargs)
        mask = nodes[self.mask_node_attr_name].squeeze()
>       return attr_values * mask
E       RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```
## What issue or task does this change relate to?

No issue created

##  Additional notes ##

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

---------

Co-authored-by: Aaron Hopkinson <aaron.hopkinson@ecmwf.int>

b8raoult committed:
docs: downloading era5 o96 dataset (#307)

## Description

<!-- Provide a brief summary of the changes introduced in this pull
request. -->

## Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
-   [x] Documentation update

## Issue Number

<!-- Link the Issue number this change addresses, ideally in one of the
"magic format" such as Closes #XYZ -->

<!-- Alternatively, explain the motivation behind the changes and the
context in which they are being made. -->

## Code Compatibility

-   [ ] I have performed a self-review of my code

### Code Performance and Testing

- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I ran the [complete Pytest
test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html)
suite locally, and they pass
-   [ ] I have tested the changes on a single GPU
-   [ ] I have tested the changes on multiple GPUs / multi-node setups
- [ ] I have run the Benchmark Profiler against the old version of the
code
- [ ] If the new feature introduces modifications at the config level, I
have made sure to update Pydantic Schemas and default configs
accordingly


<!-- In case this affects the model sharding or other specific
components please describe these here. -->

### Dependencies

- [ ] I have ensured that the code is still pip-installable after the
changes and runs
- [ ] I have tested that new dependencies themselves are
pip-installable.
- [ ] I have not introduced new dependencies in the inference portion of
the pipeline

<!-- List any new dependencies that are required for this change and the
justification to add them. -->

### Documentation

-   [ ] My code follows the style guidelines of this project
- [ ] I have updated the documentation and docstrings to reflect the
changes
- [ ] I have added comments to my code, particularly in
hard-to-understand areas

<!-- Describe any major updates to the documentation -->

## Additional Notes

<!-- Include any additional information, caveats, or considerations that
the reviewer should be aware of. -->


<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚:
https://anemoi-training--307.org.readthedocs.build/en/307/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚:
https://anemoi-graphs--307.org.readthedocs.build/en/307/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚:
https://anemoi-models--307.org.readthedocs.build/en/307/

<!-- readthedocs-preview anemoi-models end -->

---------

Co-authored-by: Harrison Cook <Harrison.cook@ecmwf.int>
Co-authored-by: Aaron Hopkinson <aaron.hopkinson@ecmwf.int>

VeraChristina committed:
ci: add warning to release-please PR body (#343)

## Description
This adds a warning to the release-please PR body to not modify parts of
the release PR that are important for the release automation to work.

anaprietonem committed:
fix: update defaults for mlflow logging (#333)

## Description
<!-- What issue or task does this change relate to? -->
PR to update the default settings regarding tracking of system metrics.
New default is to set those to `False` so that if the wants to track
them, they will need to activate the system metrics by updating this to
True.

## What problem does this change solve?
<!-- Describe if it's a bugfix, new feature, doc update, or breaking
change -->
Default interval for logging system metrics was every 10 seconds. This
lead to a large amount of information being logged into the mlflow
server.


## What issue or task does this change relate to?
<!-- link to Issue Number -->
https://github.com/ecmwf/anemoi-core/issues/332

##  Additional notes ##
<!-- Include any additional information, caveats, or considerations that
the reviewer should be aware of. -->

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

---------

Co-authored-by: Gert Mertes <13658335+gmertes@users.noreply.github.com>

mchantry committed:
fix(models,training): Remove unnecessary torch-geometric maximum version (#326)

## Description

Remove unnecessary torch-geometric maximum version as this limit is no
longer required.

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

<!-- Link the Issue number this change addresses, ideally in one of the
"magic format" such as Closes #XYZ -->

<!-- Alternatively, explain the motivation behind the changes and the
context in which they are being made. -->

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

- [ ] I have added tests that prove my fix is effective or that my
feature works
- [x] I ran the [complete Pytest
test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html)
suite locally, and they pass
-   [x] I have tested the changes on a single GPU
-   [x] I have tested the changes on multiple GPUs / multi-node setups
- [x] I have run the Benchmark Profiler against the old version of the
code
- [ ] If the new feature introduces modifications at the config level, I
have made sure to update Pydantic Schemas and default configs
accordingly


<!-- In case this affects the model sharding or other specific
components please describe these here. -->

### Dependencies

- [ ] I have ensured that the code is still pip-installable after the
changes and runs
- [ ] I have tested that new dependencies themselves are
pip-installable.
- [ ] I have not introduced new dependencies in the inference portion of
the pipeline

<!-- List any new dependencies that are required for this change and the
justification to add them. -->

### Documentation

-   [ ] My code follows the style guidelines of this project
- [ ] I have updated the documentation and docstrings to reflect the
changes
- [ ] I have added comments to my code, particularly in
hard-to-understand areas

<!-- Describe any major updates to the documentation -->

## Additional Notes

<!-- Include any additional information, caveats, or considerations that
the reviewer should be aware of. -->

---------

VeraChristina committed:
test: add training cycle test without config validation (#337)

## Description
Add a training cycle test to test if training with the basic global
config works when running without config validation.

## What problem does this change solve?
All other integration tests in training use config validation. This is
just to test one use case without config validation as we still provide
this option.

## What issue or task does this change relate to?
Integration tests in training

##  Additional notes ##

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

VeraChristina committed:
test: add test for lam training with existing graph (#336)

## Description
This adds an integration tests in training to run a test cycle with a
lam grid and an existing graph.

## What problem does this change solve?
We test if training works when loading a graph from disk. This is a
feature used often for lam/stretched grid use cases.

## What issue or task does this change relate to?
Integration tests for training

##  Additional notes ##
We are looking into a better testing configuration for the graphs used
in lam/stretched grids separately. This is just to test the
functionality of loading a graph during training.

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

VeraChristina committed:
ci: add notify team step to hpc workflow (#344)

## Description
This PR changes the ci workflow for integration tests on the hpc.
- It adds a notify-team step, as for the integration test workflow on
github runners, to notify the AnemoiSecurity group if tests fail --
currently only the person who set up the cron job gets a notification.
 - It makes the pytest output more verbose.

DeployDuck committed:
chore(main): Release models 0.7.0 (#324)

:robot: Automated Release PR

This PR was created by `release-please` to prepare the next release.
Once merged:

1. A new version tag will be created
2. A GitHub release will be published
3. The changelog will be updated

Changes to be included in the next release:
---


##
[0.7.0](https://github.com/ecmwf/anemoi-core/compare/models-0.6.0...models-0.7.0)
(2025-05-30)


### ⚠ BREAKING CHANGES

* generalise activation function
([#163](https://github.com/ecmwf/anemoi-core/issues/163))

### Features

* generalise activation function
([#163](https://github.com/ecmwf/anemoi-core/issues/163))
([98c4d06](https://github.com/ecmwf/anemoi-core/commit/98c4d06dfb5b79f605fab885c67a8a4cd6d35384))
* transformer mapper
([#179](https://github.com/ecmwf/anemoi-core/issues/179))
([2cea7db](https://github.com/ecmwf/anemoi-core/commit/2cea7db51d5c5ef63bb4b9c266deb05fb2acf66f))


### Bug Fixes

* **models,training:** Remove unnecessary torch-geometric maximum
version ([#326](https://github.com/ecmwf/anemoi-core/issues/326))
([fe93ea8](https://github.com/ecmwf/anemoi-core/commit/fe93ea8feb379147a9f9e5c5358ea8144855dc77))
* remove activation entry from MLP noise block
([#340](https://github.com/ecmwf/anemoi-core/issues/340))
([2d060f5](https://github.com/ecmwf/anemoi-core/commit/2d060f5e3382454b06c6369141942b8d6367fb4b))

---
> [!IMPORTANT]
> Please do not change the PR title, manifest file, or any other
automatically generated content in this PR unless you understand the
implications. Changes here can break the release process.
> 
> :warning: Merging this PR will:
> - Create a new release
> - Trigger deployment pipelines
> - Update package versions

 **Before merging:**
 - Ensure all tests pass
 - Review the changelog carefully
 - Get required approvals

[Release-please
documentation](https://github.com/googleapis/release-please)

sahahner committed:
revert(model,training)!: Remove multimapper (#268)

## Description

**Problem:**
For the multimapping (for example mwd to sin(mwd) and cos(mwd)), we need
to create space for the new columns where the variables are mapped to
Suboptimal solution on main:
- create a new batch
- this reallocates also the columns that are not touched
- we need memory for two batches! this does not necessarily cause
problems at o96 and n320 but leads to problems at 9km scale since the
batch takes up ~8GB

**Solution:**
Anemoi-inference now gives the flexibility to allow multimappings
compared to a few months ago, when this was more complex. During dataset
building the multimapping will be done by anemoi-transform
https://github.com/ecmwf/anemoi-transform/pull/31.

Issue https://github.com/ecmwf/aifs-dev/issues/25 shows nicely how the
model needs less memory since the batch is not copied by the
preprocessing in validation anymore.

**Breaking Change**
If the multimapper is used, this needs to be moved to the dataset
creation as the multimapping in the model will not be supported anymore
because it leads to unnecessary memory usage.
**Monomapper is still supported!** (introduced in PR
https://github.com/ecmwf/anemoi-models/pull/88) The naming `monomapper`
will disappear, and we will refer to it as `remapper` only without
distinguishing between mono and multimapping anymore.

**Config file changes**
`data.remapped` needs to be removed from all config files, as this
remapping functionality is no longer supported by the data indices.

## Type of Change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
-   [x] Documentation update

## Issue Number

Closes https://github.com/ecmwf/aifs-dev/issues/25

## Code Compatibility

-   [x] I have performed a self-review of my code

### Code Performance and Testing

- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] I ran the [complete Pytest
test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html)
suite locally, and they pass
-   [x] I have tested the changes on a single GPU
-   [x] I have tested the changes on multiple GPUs / multi-node setups
- [x] I have run the Benchmark Profiler against the old version of the
code
- [x] If the new feature introduces modifications at the config level, I
have made sure to update Pydantic Schemas and default configs
accordingly

### Dependencies

- [x] I have ensured that the code is still pip-installable after the
changes and runs
- [ ] I have tested that new dependencies themselves are
pip-installable.
- [x] I have not introduced new dependencies in the inference portion of
the pipeline

### Documentation

-   [x] My code follows the style guidelines of this project
- [x] I have updated the documentation and docstrings to reflect the
changes
- [x] I have added comments to my code, particularly in
hard-to-understand areas

## Additional Notes

Should not be merged before changes are in main branch of
anemoi-inference:
https://github.com/ecmwf/anemoi-inference/tree/fix/waves

Possible reviewers: @japols 


<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚:
https://anemoi-training--268.org.readthedocs.build/en/268/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚:
https://anemoi-graphs--268.org.readthedocs.build/en/268/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚:
https://anemoi-models--268.org.readthedocs.build/en/268/

<!-- readthedocs-preview anemoi-models end -->

mc4117 committed:
fix: dataset_order (#334)

Currently if load_weights_only=True, the order of the variables in the
dataset pre and post loading the checkpoint is not being checked. This
change fixes this bug

This has already been flagged in issue #224

pre-commit-ci[bot] committed:
chore(deps): pre-commit.ci autoupdate (#347)

<!--pre-commit.ci start-->
updates:
- [github.com/astral-sh/ruff-pre-commit: v0.11.4 →
v0.11.12](https://github.com/astral-sh/ruff-pre-commit/compare/v0.11.4...v0.11.12)
- [github.com/tox-dev/pyproject-fmt: v2.5.1 →
v2.6.0](https://github.com/tox-dev/pyproject-fmt/compare/v2.5.1...v2.6.0)
<!--pre-commit.ci end-->

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

Magnus-SI committed:
fix(training, models): Update interpolator to work with new features (#322)

## Description

This fixes a few breaking issues with the interpolator after the
addition of new features:
- Added truncation_data and statsitics tendencies arguments.
- Added handling of truncation data to the interpolator in models.
- Removed second declaration of input_dim in encoder_processor_decoder.
This is already declared in calculate_shapes_and_indices, and the second
declaration overrides the custom one in the interpolator.

## Type of Change

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
-   [ ] Documentation update

## Code Compatibility

-   [ ] I have performed a self-review of my code

### Code Performance and Testing

- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I ran the [complete Pytest
test](https://anemoi.readthedocs.io/projects/training/en/latest/dev/testing.html)
suite locally, and they pass
-   [x] I have tested the changes on a single GPU
-   [x] I have tested the changes on multiple GPUs / multi-node setups
- [ ] I have run the Benchmark Profiler against the old version of the
code
- [ ] If the new feature introduces modifications at the config level, I
have made sure to update Pydantic Schemas and default configs
accordingly

### Dependencies

-  No new dependencies

<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚:
https://anemoi-training--322.org.readthedocs.build/en/322/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚:
https://anemoi-graphs--322.org.readthedocs.build/en/322/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚:
https://anemoi-models--322.org.readthedocs.build/en/322/

<!-- readthedocs-preview anemoi-models end -->

---------

Co-authored-by: Gert Mertes <13658335+gmertes@users.noreply.github.com>

gmertes committed:
ci: Synchronised releases (#353)

We currently release each core package separately. At the repo level the
tests ensure compatibility of the state of main, but in the case of
breaking changes a person has to decide which core packages to release
together so that pypi releases remain compatible.

It has happened a few times now where we released models without
realising it broke the pypi version of training.

This PR turns on synchronised releases: instead of 3 PRs, there will now
be 1 PR to trigger a release of all packages that have been changed.
Each core package will still have its own independent version number,
and it should(tm) not create a release for a package if it has not
changed since the last release.

The result will be that every latest pypi release mirrors the state of
main at the time of release, thus ensuring compatibility between all
core packages.

gmertes committed:
ci: Remove release tests, test main for release (#361)

## Description
Now that we release all core packages at the same time, the release test
where we test with dependencies installed from pypi is no longer needed.
We just have to test main before releasing.

This PR removes the old workflow and re-enables the regular test
workflow on release PRs.

anaprietonem committed:
fix(deps): correct anemoi-models dependency version (#351)

DeployDuck committed:
chore: Release main (#360)

:robot: Automated Release PR

This PR was created by `release-please` to prepare the next release.
Once merged:

1. A new version tag will be created
2. A GitHub release will be published
3. The changelog will be updated

Changes to be included in the next release:
---


<details><summary>training: 0.5.0</summary>

##
[0.5.0](https://github.com/ecmwf/anemoi-core/compare/training-0.4.0...training-0.5.0)
(2025-06-05)


### ⚠ BREAKING CHANGES

* **models,training:** Remove multimapper
([#268](https://github.com/ecmwf/anemoi-core/issues/268))
* generalise activation function
([#163](https://github.com/ecmwf/anemoi-core/issues/163))
* Rework Loss Scalings to provide better modularity
([#52](https://github.com/ecmwf/anemoi-core/issues/52))

### Features

* Generalise activation function
([#163](https://github.com/ecmwf/anemoi-core/issues/163))
([98c4d06](https://github.com/ecmwf/anemoi-core/commit/98c4d06dfb5b79f605fab885c67a8a4cd6d35384))
* **graphs:** Add scale_resolutions arg
([#188](https://github.com/ecmwf/anemoi-core/issues/188))
([0ea0642](https://github.com/ecmwf/anemoi-core/commit/0ea06423e4979084b3afe70c30e43bb5dc5f88d5))
* **models,training:** Remove multimapper
([#268](https://github.com/ecmwf/anemoi-core/issues/268))
([0e8bb99](https://github.com/ecmwf/anemoi-core/commit/0e8bb998176bea2d653ca40772e4e6e1578551f7))
* Transformer mapper
([#179](https://github.com/ecmwf/anemoi-core/issues/179))
([2cea7db](https://github.com/ecmwf/anemoi-core/commit/2cea7db51d5c5ef63bb4b9c266deb05fb2acf66f))


### Bug Fixes

* Adapt ensemble configs to work with loss refactor.
([#300](https://github.com/ecmwf/anemoi-core/issues/300))
([c210478](https://github.com/ecmwf/anemoi-core/commit/c210478d8e84aaa2e93256aa1b4ceda74e191c7d))
* Adapt predict_step in model interface to pass on arguments for model
classes ([#281](https://github.com/ecmwf/anemoi-core/issues/281))
([a5b2643](https://github.com/ecmwf/anemoi-core/commit/a5b26432bc7b78577cd1febd5091b059cc82805c))
* Change output format in validation step to fix issue in plotting
([#305](https://github.com/ecmwf/anemoi-core/issues/305))
([cc941f3](https://github.com/ecmwf/anemoi-core/commit/cc941f3d7520e230e733654cb45e5c6473b64152))
* Clean params complement
([#291](https://github.com/ecmwf/anemoi-core/issues/291))
([4e9ca31](https://github.com/ecmwf/anemoi-core/commit/4e9ca313f838f5eccb305d6bf3b9afd8426f095f))
* Dataset_order
([#334](https://github.com/ecmwf/anemoi-core/issues/334))
([762227a](https://github.com/ecmwf/anemoi-core/commit/762227a5a25843dd4531eef1a9cbe86516eaffcd))
* **deps:** Bump `anemoi-models >= 0.8`
([#351](https://github.com/ecmwf/anemoi-core/issues/351))
([1fbd525](https://github.com/ecmwf/anemoi-core/commit/1fbd5252ef1400eb4bbe5bfbbb171dad1d652c63))
* **models,training:** Remove unnecessary torch-geometric maximum
version ([#326](https://github.com/ecmwf/anemoi-core/issues/326))
([fe93ea8](https://github.com/ecmwf/anemoi-core/commit/fe93ea8feb379147a9f9e5c5358ea8144855dc77))
* Move scaler to device in scale
([#317](https://github.com/ecmwf/anemoi-core/issues/317))
([1592d09](https://github.com/ecmwf/anemoi-core/commit/1592d09f0915dddba7dcfc1c11897e0cae5cb6d0))
* Remove activation entry from MLP noise block
([#340](https://github.com/ecmwf/anemoi-core/issues/340))
([2d060f5](https://github.com/ecmwf/anemoi-core/commit/2d060f5e3382454b06c6369141942b8d6367fb4b))
* Rework Loss Scalings to provide better modularity
([#52](https://github.com/ecmwf/anemoi-core/issues/52))
([162b906](https://github.com/ecmwf/anemoi-core/commit/162b9062882c321a4a265b0cf561be3f141ac97a))
* **training, models:** Update interpolator to work with new features
([#322](https://github.com/ecmwf/anemoi-core/issues/322))
([cfdc99f](https://github.com/ecmwf/anemoi-core/commit/cfdc99f984f0038b16cb96d73d02a25284af717e))
* **training:** Bump anemoi-graphs version to 0.5.2
([#276](https://github.com/ecmwf/anemoi-core/issues/276))
([9b8ec13](https://github.com/ecmwf/anemoi-core/commit/9b8ec13b56f0a18ac887c754d5d95a7953b2625d))
* **training:** Explicit Batch invariance
([#318](https://github.com/ecmwf/anemoi-core/issues/318))
([45f6e15](https://github.com/ecmwf/anemoi-core/commit/45f6e15183dbf61ad259b7ca5ed35d94395d237c))
* **training:** No logging variable metadata to mlflow
([#304](https://github.com/ecmwf/anemoi-core/issues/304))
([4f7c2c9](https://github.com/ecmwf/anemoi-core/commit/4f7c2c9c75ae6e67a148699a317528cd7651bf46))
* Update defaults for mlflow logging
([#333](https://github.com/ecmwf/anemoi-core/issues/333))
([5560581](https://github.com/ecmwf/anemoi-core/commit/556058167439ae29e6c9559d67f2e01df8466158))


### Documentation

* Downloading era5 o96 dataset
([#307](https://github.com/ecmwf/anemoi-core/issues/307))
([b9e2d24](https://github.com/ecmwf/anemoi-core/commit/b9e2d24702132e43f160a428ccad88b7584fcabe))
* Fix minor mistakes in CRPS user guide.
([#264](https://github.com/ecmwf/anemoi-core/issues/264))
([52d9a2e](https://github.com/ecmwf/anemoi-core/commit/52d9a2e5e0cef82b65b604ed2dc55016a48aeaa2))
</details>

<details><summary>graphs: 0.6.1</summary>

##
[0.6.1](https://github.com/ecmwf/anemoi-core/compare/graphs-0.6.0...graphs-0.6.1)
(2025-06-05)


### Bug Fixes

* **graphs:** Fix device mismatch error
([#339](https://github.com/ecmwf/anemoi-core/issues/339))
([8712c2a](https://github.com/ecmwf/anemoi-core/commit/8712c2abdd823541834b17bcdb92bd13bac101ee))
</details>

<details><summary>models: 0.8.0</summary>

##
[0.8.0](https://github.com/ecmwf/anemoi-core/compare/models-0.7.0...models-0.8.0)
(2025-06-05)


### ⚠ BREAKING CHANGES

* **models,training:** Remove multimapper
([#268](https://github.com/ecmwf/anemoi-core/issues/268))

### Features

* **models,training:** Remove multimapper
([#268](https://github.com/ecmwf/anemoi-core/issues/268))
([0e8bb99](https://github.com/ecmwf/anemoi-core/commit/0e8bb998176bea2d653ca40772e4e6e1578551f7))


### Bug Fixes

* Dataset_order
([#334](https://github.com/ecmwf/anemoi-core/issues/334))
([762227a](https://github.com/ecmwf/anemoi-core/commit/762227a5a25843dd4531eef1a9cbe86516eaffcd))
* **training, models:** Update interpolator to work with new features
([#322](https://github.com/ecmwf/anemoi-core/issues/322))
([cfdc99f](https://github.com/ecmwf/anemoi-core/commit/cfdc99f984f0038b16cb96d73d02a25284af717e))
</details>

---
> [!IMPORTANT]
> Please do not change the PR title, manifest file, or any other
automatically generated content in this PR unless you understand the
implications. Changes here can break the release process.
> 
> :warning: Merging this PR will:
> - Create a new release
> - Trigger deployment pipelines
> - Update package versions

 **Before merging:**
 - Ensure all tests pass
 - Review the changelog carefully
 - Get required approvals

[Release-please
documentation](https://github.com/googleapis/release-please)

JPXKQX committed:
feat(graphs): add GridsMask node attribute builder (#355)

## Description
Add a new node attribute builder, `GridsMask`.

## What problem does this change solve?
This PR enables more customisation for regional models.

## What issue or task does this change relate to?
Issue #350 

---------
Co-authored-by: Sabrina Wahl <sabrina.wahl@dwd.de>

JPXKQX committed:
fix(graphs): move libraries to optional dependencies (#338)

## Description
This PR moves some dependencies to optional.

## What problem does this change solve?
Some users are experiencing issues when installing `h3` via `pip install
anemoi-graphs`. This is an optional library that is only used with
`HexNodes`. We are moving this and other libraries to the optional
dependencies __extra__ category.

## What issue or task does this change relate to?
Issue #331 


---------

Co-authored-by: Aaron Hopkinson <aaron.hopkinson@ecmwf.int>

ssmmnn11 committed:
fix(training, losses): NaN handling for crps (#358)

## Description
Introduce NaN compatibility for kcrps losses

## What problem does this change solve?
kcrps losses were not compatible with NaNs

## What issue or task does this change relate to?
<!-- link to Issue Number -->

##  Additional notes ##
ack @sahahner for testing

floriankrb committed:
fix: do not inherit from a concrete class (#359)

## Description
The code was inheriting from a concrete class.

## What problem does this change solve?
This implements better inheritance.

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

floriankrb committed:
fix: revert PR 359 (#365)

theissenhelen committed:
fix: config validator checks for conflict with bounding and graphtran… (#352)

## Description
This PR introduces a new validator in the config to check whether the
config.model.decoder.initialise_data_extractor_zero = True is not used
with any bounding.

Closes #346 

## What problem does this change solve?
This is mainly a concern for the GraphTransformer in the decoder, see
issue #346.

## What issue or task does this change relate to?
#346

##  Additional notes ##
<!-- Include any additional information, caveats, or considerations that
the reviewer should be aware of. -->

***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

---------

Co-authored-by: Jakob Schloer <jakob.schloer@gmail.com>

japols committed:
feat(models,training): Shard everything (#121)

Improved scalability for large input grid sizes by making the whole
pipeline from loading the batch to computing the loss fully sharded,
never materialising the whole input/output grid in GPU memory. This also
helps reduce memory usage for long rollouts at lower resolutions.

When using the config option `model.keep_batch_sharded=True` together
with reading the batch in shards via
`dataloader.read_group_size=hardware.num_gpus_per_model`, the batch is
handed to the model already sharded across the model_comm_group and the
model returns it's prediction in shards. The loss is computed locally on
each shard to then get the global loss via all-reduce.

9km training step with 16 gpus per model, left:
keep_batch_sharded=False, right: True

<img width="1494" alt="Screenshot 2025-02-04 at 20 10 58"
src="https://github.com/user-attachments/assets/cb650be5-1c28-41b8-9209-9d94334d4924"
/>

Note that keeping the batch sharded is not supported for the
Hierarchical model as it's currently broken on main. Support for
hierarchical will be added in a separate PR as soon as it is fixed.

<!-- readthedocs-preview anemoi-training start -->
----
📚 Documentation preview 📚:
https://anemoi-training--121.org.readthedocs.build/en/121/

<!-- readthedocs-preview anemoi-training end -->

<!-- readthedocs-preview anemoi-graphs start -->
----
📚 Documentation preview 📚:
https://anemoi-graphs--121.org.readthedocs.build/en/121/

<!-- readthedocs-preview anemoi-graphs end -->

<!-- readthedocs-preview anemoi-models start -->
----
📚 Documentation preview 📚:
https://anemoi-models--121.org.readthedocs.build/en/121/

<!-- readthedocs-preview anemoi-models end -->

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Simon Lang <simon.lang@ecmwf.int>
Co-authored-by: Cathal OBrien <cathal.obrien@ecmwf.int>

matschreiner committed:
feat(graphs): build nodes from xarray compatible data (#330)

## Description

This PR adds an XArray node builder for building nodes from files
compatible with xarray.

Example usage: 
```
    data_nodes = XArrayNodes("/storage/danra/height_levels.zarr", "data", lat_key='latitude', lon_key='longitude')
    graph = HeteroData()
    graph = data_nodes.update_graph(graph)
```
Builds a set of nodes on the latitudes and longitudes specified in the
file. Therefore this builders requires latitudes and longitudes to be
defined on the file. The names of these can be specified with the
`lat/lon_key` kwargs, defaulting to `lat` and `lon`, such that xarray
can access them with its `__getitem__` method on the instantiated
`<xarrray.Dataset>`.

Additionally, since this work is related to zarr, some mock objects
(that were anemoi-dataset rather than zarr mocks) called for example
`mock_zarr_dataset`, have been renamed to `mock_anemoi_dataset`, such
that the `mock_zarr_dataset` is made available for actual zarr dataset
mocks in the test suite.

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Mario Santa Cruz <48736305+JPXKQX@users.noreply.github.com>

havardhhaugen committed:
fix: SphericalAreaWeights (#363)

## Description
Fixes #362 

## What problem does this change solve?
SphericalAreaWeights needs float64 precision input to work as intended
for high resolution graphs

## What issue or task does this change relate to?
#362 

##  Additional notes ##


***As a contributor to the Anemoi framework, please ensure that your
changes include unit tests, updates to any affected dependencies and
documentation, and have been tested in a parallel setting (i.e., with
multiple GPUs). As a reviewer, you are also responsible for verifying
these aspects and requesting changes if they are not adequately
addressed. For guidelines about those please refer to
https://anemoi.readthedocs.io/en/latest/***

DeployDuck committed:
chore: Release main (#364)

:robot: Automated Release PR

This PR was created by `release-please` to prepare the next release.
Once merged:

1. A new version tag will be created
2. A GitHub release will be published
3. The changelog will be updated

Changes to be included in the next release:
---


<details><summary>training: 0.5.1</summary>

##
[0.5.1](https://github.com/ecmwf/anemoi-core/compare/training-0.5.0...training-0.5.1)
(2025-06-17)


### Features

* **models,training:** Shard everything
([#121](https://github.com/ecmwf/anemoi-core/issues/121))
([06dde94](https://github.com/ecmwf/anemoi-core/commit/06dde94219119746215b767b846542ee31bbff63))


### Bug Fixes

* Config validator checks for conflict with bounding and graphtran…
([#352](https://github.com/ecmwf/anemoi-core/issues/352))
([f97313e](https://github.com/ecmwf/anemoi-core/commit/f97313e608f0ce2eebd46bad8e5300edebe6eec1))
* **training, losses:** NaN handling for crps
([#358](https://github.com/ecmwf/anemoi-core/issues/358))
([daf9ddd](https://github.com/ecmwf/anemoi-core/commit/daf9dddda8ff8755b59f0559666d6539d525ba3b))
</details>

<details><summary>graphs: 0.6.2</summary>

##
[0.6.2](https://github.com/ecmwf/anemoi-core/compare/graphs-0.6.1...graphs-0.6.2)
(2025-06-17)


### Features

* **graphs:** Add GridsMask node attribute builder
([#355](https://github.com/ecmwf/anemoi-core/issues/355))
([c4db823](https://github.com/ecmwf/anemoi-core/commit/c4db8236374b245be17319caf3dc26911800da10))
* **graphs:** Build nodes from xarray compatible data
([#330](https://github.com/ecmwf/anemoi-core/issues/330))
([3edaabb](https://github.com/ecmwf/anemoi-core/commit/3edaabb7e68a9ec89eccd5a16b258fbbf941b166))


### Bug Fixes

* **graphs:** Move libraries to optional dependencies
([#338](https://github.com/ecmwf/anemoi-core/issues/338))
([db215ce](https://github.com/ecmwf/anemoi-core/commit/db215ce4ec0b1835e02cf1418c71292904153547))
* SphericalAreaWeights
([#363](https://github.com/ecmwf/anemoi-core/issues/363))
([acca570](https://github.com/ecmwf/anemoi-core/commit/acca570ac33b53a4350d2492ceff05725f87ea0f))
</details>

<details><summary>models: 0.8.1</summary>

##
[0.8.1](https://github.com/ecmwf/anemoi-core/compare/models-0.8.0...models-0.8.1)
(2025-06-17)


### Features

* **models,training:** Shard everything
([#121](https://github.com/ecmwf/anemoi-core/issues/121))
([06dde94](https://github.com/ecmwf/anemoi-core/commit/06dde94219119746215b767b846542ee31bbff63))


### Bug Fixes

* Do not inherit from a concrete class
([#359](https://github.com/ecmwf/anemoi-core/issues/359))
([ca79375](https://github.com/ecmwf/anemoi-core/commit/ca7937590311c412b90dffad48fd8f25230fe5eb))
* Revert PR 359
([#365](https://github.com/ecmwf/anemoi-core/issues/365))
([fcebea4](https://github.com/ecmwf/anemoi-core/commit/fcebea484080c5333e8e213784372d1b13dab4c8))
</details>

---
> [!IMPORTANT]
> Please do not change the PR title, manifest file, or any other
automatically generated content in this PR unless you understand the
implications. Changes here can break the release process.
> 
> :warning: Merging this PR will:
> - Create a new release
> - Trigger deployment pipelines
> - Update package versions

 **Before merging:**
 - Ensure all tests pass
 - Review the changelog carefully
 - Get required approvals

[Release-please
documentation](https://github.com/googleapis/release-please)

cathalobrien committed:
add 16 mixed precision

cathalobrien committed:
merge

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gmertes reviewed (commented):
Something wonky going on with this branch, did you merge latest main into your branch correctly?
